### ISSUE TITLE
[DRC-490] [Bug] recce summary genarete failed

### ISSUE THREAD

**Issue by tomoki-takahashi-oisix** [DataRecce/recce#336]
### Current Behavior

I ran the recce summary command, but I encountered the following error:
```
Traceback (most recent call last):
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/recce/event/track.py", line 60, in invoke
    ret = super(TrackCommand, self).invoke(ctx)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/recce/cli.py", line 220, in summary
    output = generate_markdown_summary(ctx, summary_format=kwargs.get('format'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/recce/summary.py", line 389, in generate_markdown_summary
    graph = _build_lineage_graph(base_lineage, curr_lineage)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/recce/summary.py", line 268, in _build_lineage_graph
    graph.create_edge(parent_id, child_id, 'base')
  File "/Users/takahashi_tomoki/Documents/data-cuisine-mwaa/venv/lib/python3.11/site-packages/recce/summary.py", line 222, in create_edge
    raise ValueError(f'Parent node {parent_id} not found in graph')
ValueError: Parent node snapshot.data_cuisine_dbt.advertising_agency_dbt_snapshot not found in graph
```

 

### Expected Behavior

The recce summary command should execute successfully.

### Steps To Reproduce

```
$ recce run 
$ recce summary recce_state.json
```

### Environment

- recce: 0.21.0
- OS: MacOS 14.4.1
- Python: 3.11.9
- Data Warehouse: snowflake
- dbt: 1.7.1

<sub>[DRC-490](https://linear.app/piperider/issue/DRC-490/[bug]-recce-summary-genarete-failed)</sub>

**Comment by even-wei** [comment]
Hi @tomoki-takahashi-oisix 

Is there any sensitive info in the Recce state file?
If not, sharing the state file with us could accelerate the bug fixing process.
If it's not sharable, it's OK, we will investigate with the error message first.

**Comment by tomoki-takahashi-oisix** [comment]
As you mentioned, the Recce state file contains confidential information, so I cannot share it.

**Comment by even-wei** [comment]
related issue on Slack [https://getdbt.slack.com/archives/C05C28V7CPP/p1718145080517769](https://getdbt.slack.com/archives/C05C28V7CPP/p1718145080517769)

**Comment by popcornylu** [comment]
I can reproduce this issue if there is a [snapshot](https://docs.getdbt.com/docs/build/snapshots) used in a project.

**Comment by popcornylu** [comment]
Fixed in #339
